solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -1772,11 +1786,16 @@ private void resetIndexDirectory(CoreDescriptor dcore, ConfigSet coreConfig) {
   }
 
   /**
-   * Gets the permanent (non-transient) cores that are currently loaded.
+   * Gets all loaded cores, consistent with {@link #getLoadedCoreNames()}. Caller doesn't need to
+   * close.
+   *
+   * <p>NOTE: rather dangerous API because each core is not reserved (could in theory be closed).
+   * Prefer {@link #getLoadedCoreNames()} and then call {@link #getCore(String)} then close it.
    *
    * @return An unsorted list. This list is a new copy, it can be modified by the caller (e.g. it
-   *     can be sorted).
+   *     can be sorted). Don't need to close them.
    */
+  @Deprecated
```

Comment:
Deprecated because it's trappy -- cores are not incremented; perhaps they might be closed in the middle of a caller's processing.  It's not hard for callers to return a list of core names and fetch each calling close instead.  I didn't change most callers yet though.